### PR TITLE
Fix hosts_count to reduce hosts with no subscription

### PR DESCRIPTION
**Pull request by laviro**

### CODE REVIEW

File: lib/foreman_inventory_upload/generators/archived_report.rb [hunk 1]
```diff
@@ -16,9 +16,8 @@ def render(organization:)
               first = true
               host_batches.each do |hosts_batch|
                 slice_id = Foreman.uuid
-                hosts_count = hosts_batch.count
+                hosts_count = generate_slice(tmpdir, slice_id, hosts_batch)
                 @logger.info "Adding slice #{slice_id} with #{hosts_count} hosts"
```
Comment (ShimShtein) [inline lib/foreman_inventory_upload/generators/archived_report.rb:20]: Can you move this line before the `generate_slice` call, but without the `hosts_count`?

Comment (laviro) [reply lib/foreman_inventory_upload/generators/archived_report.rb:20]: Would this work?
```rb
@logger.info "Adding slice #{slice_id}"
hosts_count = generate_slice(tmpdir, slice_id, hosts_batch)
@logger.info "slice #{slice_id} was created with #{hosts_count} hosts"
```

Comment (ShimShtein) [reply lib/foreman_inventory_upload/generators/archived_report.rb:20]: :+1:

**Comment by ShimShtein** [comment]
Merged, thanks @laviro!

**Pull request merged**